File: 2021/2/part_2.py
with open("input.txt") as f:
    lines = f.readlines()

aim = 0
h = 0
d = 0

for l in lines:
    if l != "\n":
        a,v = l.split(" ")
        v = int(v)
        if a == "forward":
            h += v
            d += aim*v
        elif a == "down":
            aim += v
        elif a == "up":
            aim -= v
        else:
            raise ValueError(l)

print(h*d)

File: 2021/2/part_2.nim
import std/strutils

let
    data = readFile("input.txt")
    lines = data.splitLines

var h, d, aim: int

for l in lines:
    if l == "\n":
        continue

    let split = l.split(" ")

    if split.len != 2:
        continue

    let
        a = split[0]
        v = split[1].parseInt

    case a:
    of "forward":
        h += v
        d += aim*v
    of "down": aim += v
    of "up": aim -= v
    else: discard

echo h * d
